src/test/kotlin/com/terraformation/backend/accelerator/ModuleEventSearchTest.kt

```diff
@@ -0,0 +1,292 @@
+
+  private val clock = TestClock()
+  private val searchService: SearchService by lazy { SearchService(dslContext) }
+  private val searchTables = SearchTables(clock)
+
+  @BeforeEach
+  fun setUp() {
+    insertUser()
+    insertOrganization()
+    insertOrganizationUser(
+        userId = inserted.userId, organizationId = inserted.organizationId, role = Role.Admin)
+    insertOrganizationInternalTag(tagId = InternalTagIds.Accelerator)
+    insertCohort()
+    insertParticipant(cohortId = inserted.cohortId)
+    insertModule()
+
+    every { user.canReadAllAcceleratorDetails() } returns true
+    every { user.organizationRoles } returns mapOf(inserted.organizationId to Role.Admin)
+  }
+
+  @Test
+  fun `return event details`() {
+    val startTime = LocalDate.of(2024, 1, 1).atTime(8, 0, 0).toInstant(ZoneOffset.UTC)
+    val endTime = LocalDate.of(2024, 1, 1).atTime(9, 0, 0).toInstant(ZoneOffset.UTC)
+    insertEvent(
+        moduleId = inserted.moduleId,
+        eventStatus = EventStatus.NotStarted,
+        eventType = EventType.Workshop,
+        meetingUrl = "https://meet.google.com",
+        slidesUrl = "https://slides.google.com",
+        recordingUrl = "https://recording.google.com",
+        startTime = startTime,
+        endTime = endTime)
+
+    val prefix = SearchFieldPrefix(searchTables.events)
+    val fields =
+        listOf(
+                "id",
+                "status",
+                "type",
+                "startTime",
+                "endTime",
+                "meetingUrl",
+                "slidesUrl",
+                "recordingUrl",
+            )
+            .map { prefix.resolve(it) }
+
+    val expected =
+        SearchResults(
+            listOf(
+                mapOf(
+                    "id" to "${inserted.eventId}",
+                    "status" to EventStatus.NotStarted.getDisplayName(Locales.GIBBERISH),
+                    "type" to EventType.Workshop.getDisplayName(Locales.GIBBERISH),
+                    "startTime" to startTime.toString(),
+                    "endTime" to endTime.toString(),
+                    "meetingUrl" to "https://meet.google.com",
+                    "slidesUrl" to "https://slides.google.com",
+                    "recordingUrl" to "https://recording.google.com",
+                )),
+            null)
+
+    val actual = Locales.GIBBERISH.use { searchService.search(prefix, fields, NoConditionNode()) }
+
+    assertJsonEquals(expected, actual)
+  }
+
+  @Test
+  fun `returns results ordered by start time by default`() {
+    val event1 = insertEvent(startTime = Instant.ofEpochSecond(400))
+    val event2 = insertEvent(startTime = Instant.ofEpochSecond(800))
+    val event3 = insertEvent(startTime = Instant.ofEpochSecond(100))
+
+    val prefix = SearchFieldPrefix(searchTables.events)
+    val fields = listOf("id").map { prefix.resolve(it) }
+
+    val expected =
+        SearchResults(
+            listOf(
+                mapOf("id" to event3.toString()),
```

Comment:
`.toString()` is totally correct, so not saying this needs to change. Another option is `"$event3"`.